src/Mapping/Driver/XmlDriver.php

```diff
@@ -76,8 +76,7 @@ class XmlDriver extends FileDriver
         ],
     ];
 
-    /** @param string|null $fileExtension */
-    public function __construct($locator, $fileExtension = self::DEFAULT_FILE_EXTENSION)
+    public function __construct($locator, ?string $fileExtension = self::DEFAULT_FILE_EXTENSION)
```

Comment:
You can use the `$locator` type from [`doctrine/collection`](https://github.com/doctrine/persistence/blob/4fd46d131a1cb642930e9307ff36de6930fe6248/src/Mapping/Driver/FileDriver.php#L43).


```suggestion
    public function __construct(string|array|FileLocator $locator, ?string $fileExtension = self::DEFAULT_FILE_EXTENSION)
```